### PR TITLE
Add flush to arbitrary file

### PR DESCRIPTION
**Pull request by Ethanol48**
## Motivation

In issue #6 was discussed of adding public functions to change state in the db, and because of this I think it makes sense to have a way to have a cache file without the need to set it in the `db` #11 , or in the case that we want to preserve the cache from the previous session without being altered.

## Solution

Added in `flush_to(cache_path)` `JsonBlockCacheDB`  implementation a function that lets you dump the db in a determined file.

### CODE REVIEW

**Comment by Ethanol48** [comment]
Added minimal documentation.

**Comment by Ethanol48** [comment]
Pending changing to `CHANGELOG.md`

**Comment by Ethanol48** [comment]
I’m sorry for any notifications sent, I was using my phone and the application doesn’t have much support and made some mistakes, my apologies.

File: src/backend.rs [hunk 1]
```diff
@@ -656,6 +657,10 @@ impl SharedBackend {
     pub fn flush_cache(&self) {
         self.cache.0.flush();
     }
+
+    pub fn flush_cache_to(&self, cache_path: Option<PathBuf>) {
```
Comment (mattsse) [inline src/backend.rs:661]: lets make this accept PathBuf

Comment (mattsse) [inline src/backend.rs:661]: also docs can't hurt here

File: src/cache.rs [hunk 2]
```diff
@@ -403,6 +403,36 @@ impl JsonBlockCacheDB {
 
         trace!(target: "cache", "saved json cache");
     }
+
+    /// Flushes the DB to disk
+    #[instrument(level = "warn", skip_all, fields(path = ?self.cache_path))]
+    pub fn flush_to(&self, cache_path: Option<PathBuf>) {
+        let Some(path) = cache_path else {
```
Comment (mattsse) [inline src/cache.rs:410]: we should reuse this in the existing flush function so that we don't duplicate this

**Comment by Ethanol48** [comment]
Done 👍🏼

File: src/backend.rs [hunk 3]
```diff
@@ -656,6 +657,11 @@ impl SharedBackend {
     pub fn flush_cache(&self) {
         self.cache.0.flush();
     }
+
+    /// Flushes the DB to a specific file
+    pub fn flush_cache_to(&self, cache_path: PathBuf) {
```
Comment (DaniPopes) [inline src/backend.rs:662]: Should be `&Path`.

File: src/cache.rs [hunk 4]
```diff
@@ -382,6 +382,13 @@ impl JsonBlockCacheDB {
     #[instrument(level = "warn", skip_all, fields(path = ?self.cache_path))]
     pub fn flush(&self) {
         let Some(path) = &self.cache_path else { return };
+        self.flush_to(path.clone());
+    }
+
+    /// Flushes the DB to a specific file
+    pub fn flush_to(&self, cache_path: PathBuf) {
```
Comment (DaniPopes) [inline src/cache.rs:389]: Should be `&Path`.

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]
lgtm!

**Pull request merged**